### ISSUE TITLE
fork gaphql-js and use as npm dependency

### ISSUE THREAD

**Issue by keski** [LiUGraphQL/woo.sh#69]
The current graphql-server implementation patches part of the graphql-js npm library in order to support multiple transactions. This approach is not very clean and is difficult to maintain.

This ticket is about forking the graphql-js and extending it so that our patched library can be used as a regular npm dependency as:
```
"graphql": "git://github.com/LiUGraphQL/graphql-js.git#npm"
```
The forked library should be maintained separately and extend (not change) the current functionality of the graphql-js. To this end, the current approach to transactions (and the corresponding functions (`create`, `createEdge`, etc.) will have to be updated to return promises, since this is the expected result in graphql-js, and not use the async-wait pattern.

**Comment by keski** [comment]
The graphql-js project has been forked and updated according to the ticket. It is now available at [LiUGraphQL/graphql-js](https://github.com/LiUGraphQL/graphql-js). In order to use the new library, update `package.json` in the driver to use:
```
"graphql": "git://github.com/LiUGraphQL/graphql-js.git#npm"
```
instead of:
```
"graphql": "^14.2.1"
```

**Comment by keski** [comment]
See the refactored arangodb driver at [69-fork-gaphql-js-and-use-as-npm-dependency](https://github.com/LiUGraphQL/woo.sh/tree/69-fork-gaphql-js-and-use-as-npm-dependency)

@epii-1, could you have a look and provide some feedback on the structure? If anything is unclear, strange or plain wrong, please let me know. There's still (possibly) a bug with edge validation for nested creation of fields. If you encounter this, simply disable edge validation during server startup for now. I'll have another go at it tomorrow morning. I also haven't done much testing with annotations yet.

**Comment by keski** [comment]
I've made a few bug fixes and key and edge validation should now work. I've also updated `starwars-db.graphql` with required name key fields for Planet and Human and added an explicit setting of the `seed` parameter.

**Comment by keski** [comment]
The issue has been resolved in [69-fork-gaphql-js-and-use-as-npm-dependency](https://github.com/LiUGraphQL/woo.sh/tree/69-fork-gaphql-js-and-use-as-npm-dependency) and the branch has been merged with master into the new branch [develop](https://github.com/LiUGraphQL/woo.sh/tree/develop). Note that merging develop into master is on hold until the currently pending features have been adapted for the develop branch.

**Comment by epii-1** [comment]
@keski Sorry for being a bit delayed with my response (even if you got a minor response earlier by mail). Had some other things I had to do first.
Now I have looked through the changed to the driver a bit better and I think it a lot cleaner then before and the structure of creations and updates are now easier to follow. I really missed the possibility to call `createEdge` from `create` (as you have now made possible and done) as the getting the ids to the right format was not trivial before. As you now made this easy, certain changes and implementations now becomes much easier to do.
There are some minor odd things with how `getScalarsAndEnums` and `formatFixInput` are now used (you basically made having a different `formatFixInput` unnecessary, so the two functions can be merged again), but I will clean that up later. It really is just very minor things and should have no impact of the actual usage.
